geode-serialization/src/main/java/org/apache/geode/internal/serialization/filter/GlobalSerialFilterConfiguration.java

```diff
@@ -38,7 +38,9 @@
   private final SerializableObjectConfig serializableObjectConfig;
   private final FilterPatternFactory filterPatternFactory;
   private final Supplier<Set<String>> sanctionedClassesSupplier;
-  private final Consumer<String> logger;
+  private final Consumer<String> infoLogger;
+  private final Consumer<String> warnLogger;
+  private final Consumer<String> errorLogger;
```

Comment:
Would it be cleaner and still testable to use a single log4j-api Logger interface instead of three Consumer? The tests need to differentiate between three different log levels: INFO, WARN and ERROR.

Reply:
I think so, for several reasons:
- This API locks the implementation into accepting only strings. At some point, the implementation may want to use a richer logging API. (On the other hand: YAGNI.)
- The name `accept()` feels oddly passive, as if code is politely requesting that the logger please consider doing a thing. A logging API would make it clear that the code is telling the logger to log something.
- A logging API allows a test to observe all of the same interactions.

Reply:
Fixed